app/validators/web_push_key_validator.rb

```diff
@@ -3,6 +3,8 @@
 class WebPushKeyValidator < ActiveModel::Validator
   def validate(subscription)
     begin
+      # We test if the subscription key is a valid p256 pubkey, we
+      # use Webpush but it would work with LegacyWebpush too (it also uses the same curve)
       Webpush::Encryption.encrypt('validation_test', subscription.key_p256dh, subscription.key_auth)
```

Comment:
@p1gp1g / @ClearlyClaire perhaps we could add a `Webpush.keys_valid?(key_p256_dh, key_auth) #=> boolean` method? That way we're not actually messing about with the encryption here, but instead doing exactly what we intend.

The only reason I wrote this code this way was because it was the quickest way to ship _some_ validation, without reimplementing a bunch of stuff.

Reply:
Trying to encrypt something looks like a good way to test the value to me :+1:

Reply:
I'm not against a more specific test, but it does not need to be in this PR.